### ISSUE TITLE
Manager language

### ISSUE THREAD

**Issue by mplavala** [evolution-cms/evolution#1934]
When changing System configuration, the Manager language field is pre-filled with random language (different for different sites). This is very bad, because if you save new settings, the language of manager randomly changes, some of the changes cause bugs of other plugins (e.g. arabic language causes problems to ElementsInTree). This is quite annoying.
For it to work properly, Evolution should pre-select the current language, not a random one.

**Comment by BBloke** [comment]
I posted the same thing although it defaults to BG for me.  I closed the issue as I didn't think it was applicable to others.  Just something on my setup.  It's nice to know I'm not alone.

Here's the original post I made: https://github.com/evolution-cms/evolution/issues/1932

**Comment by mplavala** [comment]
> it defaults to BG for me.

The default is random. All my sites have (should have) English as Manager language.

I have one site, where default on production server is BG, while the local development version in my computer (via LAMP) defaults to FA (which is Arabic, so right-to-left, lot of fun).
Other site production has default EN, but local development again defaults to FA.
Third development site (without production version, this is just my toy site, but it was a fresh 3.0.x install) defaults to EN, as it should.

All dev sites are on my computer, running under the same PHP and database, yet results vary.

I tried changing language to other option on one site (CS instead of EN), but the default was still the same as before and different from what it should have been. This was on a **site that was updated from 2.0.x** and 1.4.x before.
I also tried changing the language on a **fresh 3.1.x install and the default changed** as it should.

Therefore **I think that this problem only affects sites that were updated from 2.0.x to 3**. Also the order of the languages in the selection seems to be random (maybe issue there?).

**Comment by leonie12** [comment]
Also here with a site updated from 2.0.4 to 3.1: I choose for example EN as language. Everytime I change settings (not the language) in manager and save it the language reverts to RU.

**Comment by helgispbru** [comment]
Проблема повторяется в 3.1.6

**Comment by mplavala** [comment]
I found the problem and a quick-fix. But proper solution should also be considered by the devs.

TLDR; Go to Users, select your user. In settings to the second tab (it is also called User), select Manager language to be blank and save. Repeat for every other user. Problem solved.

What happens here: in 1.4.x branch the selected language was stored in the database as "english" or "french" and in 3.x branch the language is stored as "en" or "fr". When updating from 1.4.x to 3.x the update script correctly handles global language settings, but they forgot about user language settings. Also (for some reason unknown to me, and this is probably a bug) when the system asks you to select a language and you have some language selected for your user, the system preselects the users language. This is quite unexpected behaviour and this also causes the issue as follows: you user has saved manager language as "english" because you did this back on the 1.4.x branch, while Evo compares this to "en" when it is deciding whether to preselect it or not. Since "english" != "en", no language is preselected, in which case first option is preselected (and this is likely not what you want).

The obvious solution would be to handle user-selected language vs. global-selected language. But also I observed that the user-selected language does nothing, so maybe it will get removed once and this issue will stop being a problem.

Anyways, I feel like we can close this now, since I am not sure whether this is a bug or a feature and you most likely have to fix this yourself in your system.

**Comment by fourroses666** [comment]
My users manager language was blank. 
After resaving the users it seem to remember the manager language on the Configuration page.
